frame/asset-conversion/src/lib.rs

```diff
@@ -778,6 +760,13 @@ pub mod pallet {
 					}
 					i.saturating_inc();
 				}
+				Self::deposit_event(Event::SwapExecuted {
+					who: sender,
+					send_to,
+					path,
+					amount_in: *first_amount,
+					amount_out: *amounts.last().expect("Always has more than 1 element"),
```

Comment:
We don't ensure this anywhere, we're just assuming it will be used this way. I think it's probably better to emit an error, same for `first_amount`.
```rust
let first_amount = amounts.first().ok_or(Error::<T>::PathError)?;
let amount_out = amounts.last().ok_or(Error::<T>::PathError)?;
```

Reply:
https://github.com/paritytech/substrate/blob/074dc1dfee4132b597db6eb147a86c72b751f03c/frame/asset-conversion/src/lib.rs#L848
and 
https://github.com/paritytech/substrate/blob/074dc1dfee4132b597db6eb147a86c72b751f03c/frame/asset-conversion/src/lib.rs#L867
both ensure that the functions will return with a minimum of one item in the vec. That said the path validation checks (mentioned earlier) ensure that there will be a minimum of two elements in the vec. Would it make it clearer if I add an assertion on the results of `get_amounts_out` and `get_amounts_in` that the length was at least 2?

Reply:
I've added an `ensure` to enforce that amounts is always length 2 or more.